File: numbers-of-letters-of-numbers/__tests__/numbersOfLettersOfNumbers.test.js
const { numbersOfLetters } = require('../numbersOfLettersOfNumbers');

describe(`If we write out the digits of "60" as English words we get "sixzero"; the number of letters in "sixzero" is seven. The number of letters in "seven" is five. The number of letters in "five" is four. The number of letters in "four" is four: we have reached a stable equilibrium.
    Note: for integers larger than 9, write out the names of each digit in a single word (instead of the proper name of the number in English). For example, write 12 as "onetwo" (instead of twelve), and 999 as "nineninenine" (instead of nine hundred and ninety-nine).
    For any integer between 0 and 999, return an array showing the path from that integer to a stable equilibrium:`, () => {
    test('Given 60, return ["sixzero", "seven", "five", "four"]', () => {
        const number = 60;
        expect(numbersOfLetters(number)).toEqual(["sixzero", "seven", "five", "four"]);
    });
    test('Given 12, return ["onetwo", "six", "three", "five", "four"]', () => {
        const number = 12;
        expect(numbersOfLetters(number)).toEqual(["onetwo", "six", "three", "five", "four"]);
    });
    test('Given 37, return ["threeseven", "onezero", "seven", "five", "four"]', () => {
        const number = 37;
        expect(numbersOfLetters(number)).toEqual(["threeseven", "onezero", "seven", "five", "four"]);
    });
    test('Given 311, return ["threeoneone", "oneone", "six", "three", "five", "four"]', () => {
        const number = 311;
        expect(numbersOfLetters(number)).toEqual(["threeoneone", "oneone", "six", "three", "five", "four"]);
    });
    test('Given 999, return ["nineninenine", "onetwo", "six", "three", "five", "four"]', () => {
        const number = 999;
        expect(numbersOfLetters(number)).toEqual(["nineninenine", "onetwo", "six", "three", "five", "four"]);
    });
    test('Given 648, return ["sixfoureight", "onetwo", "six", "three", "five", "four"]', () => {
        const number = 648;
        expect(numbersOfLetters(number)).toEqual(["sixfoureight", "onetwo", "six", "three", "five", "four"]);
    });
    test('Given 241, return ["twofourone", "onezero", "seven", "five", "four"]', () => {
        const number = 241;
        expect(numbersOfLetters(number)).toEqual(["twofourone", "onezero", "seven", "five", "four"]);
    });
    test('Given 866, return ["eightsixsix", "oneone", "six", "three", "five", "four"]', () => {
        const number = 866;
        expect(numbersOfLetters(number)).toEqual(["eightsixsix", "oneone", "six", "three", "five", "four"]);
    });
    test('Given 350, return ["threefivezero", "onethree", "eight", "five", "four"]', () => {
        const number = 350;
        expect(numbersOfLetters(number)).toEqual(["threefivezero", "onethree", "eight", "five", "four"]);
    });
});

// https://www.codewars.com/kata/599febdc3f64cd21d8000117/train/javascript

File: python-venv/numbers_of_letters_of_numbers/__tests__/numbers_of_letters_of_numbers_test.py
from kata_function.numbers_of_letters_of_numbers import numbers_of_letters

def test_1():
    number = 60
    assert numbers_of_letters(number) == ["sixzero", "seven", "five", "four"]

def test_2():
    number = 12
    assert numbers_of_letters(number) == ["onetwo", "six", "three", "five", "four"]

def test_3():
    number = 37
    assert numbers_of_letters(number) == ["threeseven", "onezero", "seven", "five", "four"]

def test_4():
    number = 311
    assert numbers_of_letters(number) == ["threeoneone", "oneone", "six", "three", "five", "four"]

def test_5():
    number = 999
    assert numbers_of_letters(number) == ["nineninenine", "onetwo", "six", "three", "five", "four"]

def test_6():
    number = 648
    assert numbers_of_letters(number) == ["sixfoureight", "onetwo", "six", "three", "five", "four"]

def test_7():
    number = 241
    assert numbers_of_letters(number) == ["twofourone", "onezero", "seven", "five", "four"]

def test_8():
    number = 866
    assert numbers_of_letters(number) == ["eightsixsix", "oneone", "six", "three", "five", "four"]

def test_9():
    number = 350
    assert numbers_of_letters(number) == ["threefivezero", "onethree", "eight", "five", "four"]

def test_10():
    number = 4
    assert numbers_of_letters(number) == ["four"]

    
#https://www.codewars.com/kata/599febdc3f64cd21d8000117/train/python